php_version_resolver_test.go

```diff
@@ -0,0 +1,181 @@
+package composer_test
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/paketo-buildpacks/composer"
+	"github.com/sclevine/spec"
+
+	. "github.com/onsi/gomega"
+)
+
+func testPhpVersionResolver(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+
+		workingDir string
+
+		phpVersionResolver composer.PhpVersionResolver
+	)
+
+	it.Before(func() {
+		var err error
+		workingDir, err = os.MkdirTemp("", "working-dir")
+		Expect(err).NotTo(HaveOccurred())
+
+		phpVersionResolver = composer.NewPhpVersionResolver()
+	})
+
+	it.After(func() {
+		Expect(os.RemoveAll(workingDir)).To(Succeed())
+	})
+
+	context("when both composer.lock and composer.json contain php versions", func() {
+		it.Before(func() {
+			Expect(ioutil.WriteFile(filepath.Join(workingDir, "composer.json"), []byte(`{
+   "require": {
+	   "php-64bit": "php-64bit.version.from-composer-json",
+	   "php": "php-32bit.version.from-composer-json"
+   }
+}`), os.ModePerm)).NotTo(HaveOccurred())
```

Comment:
Should this be `Expect(...).To(Succeed())` or is this syntax I don't know about?

Reply:
I think I've seen both, but `To(Succeed())` seems more clear and shorter. I'll standardize on that.

Reply:
Ah I'm sorry for the lack of clarity around this. I usually would expect:
`Expect(err).NotTo(HaveOccurred())` for errors, and then `Expect(os.WriteFile(...)).To(Succeed())` when we're asserting a command we've run has worked. I realize it's a nitpick but that's the way it's done in most of the buildpacks

Reply:
Somewhat oddly, the two matchers seem functionally equivalent, just negated from each other. So it's purely for readability which to choose?

https://github.com/onsi/gomega/blob/master/matchers/succeed_matcher.go
https://github.com/onsi/gomega/blob/master/matchers/have_occurred_matcher.go

Reply:
Yes readability, this is a nitpick.